device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/CustomLoggerTest.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package tests.unit.com.microsoft.azure.sdk.iot.device;
+	
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.assertThat;
+import com.microsoft.azure.sdk.iot.device.CustomLogger;
+import org.apache.log4j.*;
+import mockit.Mocked;
+import mockit.Mock;
+import mockit.MockUp;
+import org.junit.Test;
+
+/** Unit tests for CustomLogger. */
+public class CustomLoggerTest
+{
+	// Tests_SRS_CUSTOMERLOGGER_25_001: [The constructor shall create the logger instance.]
+    @Test
+    public void contructorCreatesLogger()
+	{
+        CustomLogger logger = new CustomLogger(this.getClass());
+        assertThat(logger, notNullValue());
+	}
+	
+	// Tests_SRS_CUSTOMERLOGGER_25_002: [The function shall log message for all levels.]
+    @Test
+    public void allLevelMessageLogging()
+    {
+        CustomLogger logger = new CustomLogger(this.getClass());
+        mockLogger();
+        String message = "This is INFO message";
+        logger.LogInfo(message, logger.getMethodName());
+        message = "This is DEBUG message";
+        logger.LogDebug(message, logger.getMethodName());
+        message = "This is TRACE message";
+        logger.LogTrace(message, logger.getMethodName());
+        message = "This is TRACE message";
+        logger.LogTrace(message, logger.getMethodName());
+        message = "This is WARN message";
+        logger.LogWarn(message, logger.getMethodName());
+        message = "This is FATAL message";
+        logger.LogFatal(message, logger.getMethodName());
+        message = "This is CUSTOM ERROR message";
+        logger.LogError(message, logger.getMethodName());
+        message = "This is ERROR message";
+        logger.LogError(new Throwable(message));
+    }
+
+    // Tests_SRS_CUSTOMERLOGGER_25_002: [The function shall log message for all levels.]
+    @Test
+	public void getExecutingMethodName()
+	{
+        CustomLogger logger = new CustomLogger(this.getClass());
+        String methodName = logger.getMethodName();
+        assertThat("getExecutingMethodName", is(equalTo(methodName)));
+	}
+	// Tests_SRS_CUSTOMERLOGGER_25_002: [The function shall mock CustomLogger for unit testing.]
+	private void mockLogger() {
+        new MockUp<CustomLogger>() {
+            @Mock
+            public void LogInfo(String msg, Object... params) {
```

Comment:
How about making sure that log info actually prints log at info level? Your test cases are only making sure that calls actually call the respective function. Use invocation or some other way to actually go through the real function calls